system_metrics_collector/test/system_metrics_collector/test_linux_process_cpu_measurement_node.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <memory>
+#include <fstream>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+
+#include "metrics_statistics_msgs/msg/metrics_message.hpp"
+#include "metrics_statistics_msgs/msg/statistic_data_type.hpp"
+
+#include "../../src/system_metrics_collector/linux_process_cpu_measurement_node.hpp"
+#include "../../src/system_metrics_collector/proc_cpu_data.hpp"
+#include "../../src/system_metrics_collector/utilities.hpp"
+
+#include "test_constants.hpp"
+#include "test_utilities.hpp"
+
+using metrics_statistics_msgs::msg::MetricsMessage;
+using metrics_statistics_msgs::msg::StatisticDataPoint;
+using moving_average_statistics::StatisticData;
+
+namespace
+{
+constexpr const char kTestNodeName[] = "test_measure_linux_process_cpu";
+constexpr const char kTestTopic[] = "test_process_cpu_measure_topic";
+}
+
+class TestLinuxProcessCpuMeasurementNode : public system_metrics_collector::
```

Comment:
Maybe comment that the point of this class is to expose private / protected methods for testing.

Reply:
Done.